src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -483,26 +498,28 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
             bg='background.surface.raised.accent'
             borderBottomRadius='xl'
           >
-            <Button
+            <ButtonWalletPredicate
+              isValidWallet={Boolean(isChainSupportedByWallet)}
+              isDisabled={Boolean(
+                errors.amountFieldInput ||
+                  !runeAddress ||
+                  !isValidStakingAmount ||
+                  !(isStakeFeesSuccess || isGetApprovalFeesSuccess) ||
+                  !cooldownPeriod,
+              )}
               size='lg'
               mx={-2}
               onClick={handleWarning}
-              isDisabled={
-                Boolean(errors.amountFieldInput) ||
-                !runeAddress ||
-                !isValidStakingAmount ||
-                !(isStakeFeesSuccess || isGetApprovalFeesSuccess) ||
-                !cooldownPeriod
-              }
               isLoading={isGetApprovalFeesLoading || isStakeFeesLoading}
               colorScheme={
                 Boolean(errors.amountFieldInput || errors.manualRuneAddress) ? 'red' : 'blue'
               }
             >
               {errors.amountFieldInput?.message ||
                 errors.manualRuneAddress?.message ||
+                chainNotSupportedByWalletCopy ||
```

Comment:
suggestion (for a follow-up): use react-hook-form validation here for consistency

Reply:
How would you see this? Because the wallet and stakingAsset is not even controlled by the form, his validation isn't linked by the form at all

Reply:
Just as we're doing validation for `amountFieldInput` with `amountFieldInputRules`, we could validate chain-support with `trigger()`, but yeah in the end it may be messier